### ISSUE TITLE
OpenID Connect "code_challenge_method" parameter is vulnerable to reflective XSS

### ISSUE THREAD

**Issue by flangfeldt** [FusionAuth/fusionauth-issues#354]
### Description 
The URL parameter "code_challenge_method" is vulnerable to reflective XSS. The parameter is apparently not getting validated, the rest of the OpenID Connect parameters seem to be safe.

### Steps to reproduce
1. Go to the login page of FusionAuth.
2. Add `&code_challenge_method="><script>alert("Evil%20injection")</script>` to the end of the URL in the address bar.
3. Observe the injected javascript alert.

### Expected behavior
The parameter value should either be sanitized before parsing it in the page or an error should be thrown, like it's done for the rest of the parameters.

### Platform
(Please complete the following information)
 - Device: Desktop
 - OS: macOS
 - Chrome  - 78.0.3904.70

**Comment by robotdan** [comment]
What version of FusionAuth are you running, I've been unable to recreate.

**Comment by flangfeldt** [comment]
Version 1.10.0

**Comment by robotdan** [comment]
Ah, yes. This was fixed in 1.11.0, but if you're using a theme you'll need to correct it there. 

In `_helpers.ftl` you'll find the `[@link` macro. 

At the end of that macro you'll see this broken code :

```
&code_challenge=${code_challenge!''}&code_challenge_method=${code_challenge_method!''}">
```

Modify it to look like this one:
```
&code_challenge=${(code_challenge?url)!''}&code_challenge_method=${(code_challenge_method?url)!''}&user_code=${(user_code?url)!''}">
```
